src/UnitTest/test_local_cache.cpp

```diff
@@ -1,215 +1,254 @@
-            threads[i] = std::thread(cache_writer);
-        }
-        else
-        {
-            threads[i] = std::thread(cache_reader);
-        }
-    }
-
-    for(auto& t : threads)
-    {
-        t.join();
-    }
-
-    TEST_PASSED();
-}
-
-extern void LocalCacheTests()
-{
-    local_cache_clear();
-
-    AddGetItem();
-    OverwriteCacheEntry();
-    VerifyClearCache();
-    VerifyExpiryWorks();
-    InvalidParams();
-    ThreadSafetyTest();
-}
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+#include <gtest/gtest.h>
+
+#undef NDEBUG // ensure that asserts are never compiled out
+#include <cassert>
+#include <cstdio>
+#include <cstring>
+#include <thread>
+#if defined(__LINUX__)
+#include <unistd.h>
+#else
+#include <windows.h>
+#endif
+
+#include "unit_test.h"
+#include "../local_cache.h"
+
+#if defined(__LINUX__)
+typedef bool boolean;
+#else
+#endif
+
+static time_t now()
+{
+    return time(nullptr);
+}
+
+//
+// Add an item to the cache and retrieve it.
+//
+static boolean AddGetItem()
+{
+    TEST_START();
+
+    static const std::vector<uint8_t> data = {8, 6, 7, 5, 3, 0, 9};
+    local_cache_add(__FUNCTION__, now() + 60, data.size(), data.data());
+
+    auto retrieved = local_cache_get(__FUNCTION__);
+    assert(retrieved != nullptr);
+    assert(*retrieved == data);
+
+    TEST_PASSED();
+    return true;
+}
+
+//
+// Add an item to the cache, then overwrite it
+//
+static boolean OverwriteCacheEntry()
+{
+    TEST_START();
+
+    static const std::vector<uint8_t> data1 = {1, 1, 1, 1};
+    local_cache_add(__FUNCTION__, now() + 60, data1.size(), data1.data());
+
+    static const std::vector<uint8_t> data2 = {2};
+    local_cache_add(__FUNCTION__, now() + 60, data2.size(), data2.data());
+
+    auto retrieved = local_cache_get(__FUNCTION__);
+    assert(*retrieved == data2);
+
+    TEST_PASSED();
+    return true;
+}
+
+//
+// Add an item to the cache, clear the cache, then ensure the entry is gone
+//
+static boolean VerifyClearCache()
+{
+    TEST_START();
+
+    static const uint8_t data[] = "stuff goes here";
+    local_cache_add(__FUNCTION__, now() + 60, sizeof(data), data);
+
+    assert(nullptr != local_cache_get(__FUNCTION__));
+    local_cache_clear();
+    assert(nullptr == local_cache_get(__FUNCTION__));
+
+    TEST_PASSED();
+    return true;
+}
+
+//
+// Add an item to the cache with a short expiry, then ensure it actually
+// expires.
+//
+static boolean VerifyExpiryWorks()
+{
+    TEST_START();
+
+    int expiry_seconds = 2;
+
+    // add an entry with expiry shortly in the future
+    static const uint8_t data[] = "stuff goes here";
+    local_cache_add(__FUNCTION__, now() + expiry_seconds, sizeof(data), data);
+
+    // ensure the data is there
+    assert(nullptr != local_cache_get(__FUNCTION__));
+
+    // wait for expiry, after which the data should be gone
+
+#if defined(__LINUX__)
+    sleep(expiry_seconds);
+#else
+    Sleep(expiry_seconds * 1000);
```

Comment:
I assume this is for milliseconds? i.e all other platforms supported other than LINUX use milliseconds and Linux uses seconds?

Reply:
That comes from legacy code. I think Windows is milliseconds and Linux is seconds.